aten/src/ATen/native/Convolution.cpp

```diff
@@ -577,10 +565,17 @@ at::Tensor _convolution(
   auto k = weight.ndimension();
   // mkldnn conv2d weights could have been re-ordered to 5d by
   // mkldnn_reorder_conv2d_weight
+  c10::IntArrayRef weight_sizes = weight.sizes();
   if (input_is_mkldnn && (k == input.ndimension() + 1)) {
     k = input.ndimension();
+    std::vector<int64_t> weight_sizes_mkl(k);
+    weight_sizes_mkl[0] = weight.size(0) * weight.size(1);
+    std::copy_n(
+        weight.sizes().cbegin() + 2, k - 1, weight_sizes_mkl.begin() + 1);
+    weight_sizes = c10::IntArrayRef(weight_sizes_mkl);
```

Comment:
**WOOP WOOP OWNERSHIP POLICE**. IntArrayRef is a NON-OWNING reference to the pointed to array. You only allocated `weight_sizes_mkl` inside this conditional, which means it goes dead when you exit!

Reply:
oops, thanks!